CRM/Activity/BAO/Activity.php

```diff
@@ -859,6 +827,60 @@ public static function getActivities($params, $getCount = FALSE) {
     return $getCount ? count($activities) : $activities;
   }
 
+  /**
+   * Filter the activity types to only return the ones we actually asked for
+   * Uses params['activity_type_id'] and params['activity_type_exclude_id']
+   *
+   * @param $params
+   * @returns array (Use in Activity.get API activity_type_id)
+   */
+  public static function filterActivityTypes($params) {
+    $activityTypes = array();
+
+    // If no activity types are specified, get all the active ones
+    if (empty($params['activity_type_id'])) {
+      $activityTypes = CRM_Activity_BAO_Activity::buildOptions('activity_type_id', 'get');
+    }
+
+    // If no activity types are specified or excluded, return the list of all active ones
+    if (empty($params['activity_type_id']) && empty($params['activity_type_exclude_id'])) {
+      if (!empty($activityTypes) && count($activityTypes)) {
+        return array('IN' => array_keys($activityTypes));
+      }
+      return NULL;
```

Comment:
So this would only be hit if there were NO enabled activity types configured for the site at all? (in which case only inactive ones would be displayed)

Reply:
Pretty unlikely to hit NULL yes.  As the return parameter is used as an API parameter it would return no activity types.